plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -210,35 +218,130 @@ public function set_current_screen_to_block_editor_if_wc_admin() {
 	 * Get the product editor settings.
 	 */
 	private function get_product_editor_settings() {
+		$layout_template_registry = wc_get_container()->get( BlockTemplateRegistry::class );
+		$layout_template_logger   = BlockTemplateLogger::get_instance();
+
 		$editor_settings = array();
 
-		$template_registry     = wc_get_container()->get( BlockTemplateRegistry::class );
-		$block_template_logger = BlockTemplateLogger::get_instance();
+		foreach ( $layout_template_registry->get_all_registered() as $layout_template ) {
+			$editor_settings['layoutTemplates'][] = $layout_template->to_JSON();
 
-		$block_template_logger->log_template_events_to_file( 'simple-product' );
-		$block_template_logger->log_template_events_to_file( 'product-variation' );
+			$layout_template_logger->log_template_events_to_file( $layout_template->get_id() );
+			$editor_settings['layoutTemplateEvents'][] = $layout_template_logger->get_formatted_template_events( $layout_template->get_id() );
+		}
 
-		$editor_settings['templates'] = array(
-			'product'           => $template_registry->get_registered( 'simple-product' )->get_formatted_template(),
-			'product_variation' => $template_registry->get_registered( 'product-variation' )->get_formatted_template(),
-		);
+		$product_templates = apply_filters( 'woocommerce_product_editor_product_templates', array() );
 
-		$editor_settings['templateEvents'] = array(
-			'product'           => $block_template_logger->get_formatted_template_events( 'simple-product' ),
-			'product_variation' => $block_template_logger->get_formatted_template_events( 'product-variation' ),
-		);
+		usort( $product_templates, function ( $a, $b ) {
+			return $a->get_order() - $b->get_order();
+		} );
+
+		$editor_settings['productTemplates'] = array_map( function ( $product_template ) {
+			return $product_template->to_JSON();
+		}, $product_templates );
 
 		$block_editor_context = new WP_Block_Editor_Context( array( 'name' => self::EDITOR_CONTEXT_NAME ) );
 
 		return get_block_editor_settings( $editor_settings, $block_editor_context );
 	}
 
+	/**
+	 * Get default product templates.
+	 * 
+	 * @param array @templates The initial templates.
+	 * @return array The default templates.
+	 */
+	public function get_product_templates( array $templates ) {
```

Comment:
I'd suggest renaming to `get_default_product_templates` since that's what it does. Otherwise, one might think that the returned result from this is filtered, which it isn't.

Also, this function doesn't need to be public.

We might need a wrapper function `get_product_templates()` that calls `apply_filters( 'woocommerce_product_editor_product_templates')` later on, once we introduce the REST API. But, we can do that then.